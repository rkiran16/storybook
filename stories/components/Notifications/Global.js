import { html } from "lit-html";
import { Link } from "../Link/Link";

const GlobalIcon = (type) => {
  console.log(type);
  switch (type) {
    case "warning":
      return html` <span class="material-icons"> warning_amber </span> `;
    case "success":
      return html` <span class="material-icons"> check_circle_outline </span> `;
    default:
      return html` <span class="material-icons"> error_outline </span> `;
  }
};

export const GlobalNotification = ({
  title,
  label,
  color = "black",
  icon = "default",
  href,
  linkTitle,
}) => {
  return html`
      <div
        class="alert alert-custom bg-${color} alert-dismissible d-flex align-items-center text-white rounded-0 global-notification"
        role="alert"
      >
        <div class="container d-flex position-relative">
        ${icon ? GlobalIcon(icon) : null}
        <div>
          <strong class="p-2">${title}</strong>
          <span> ${label}</span>
          <span class="ms-2 text-white">
            ${
              href
                ? html`<a class="inline text-white" href=${href}>${linkTitle}</a>`
                : null
            }
          </span>
        </div>
        <button
          type="button"
          class="btn-close btn-close-white p-1" 
          data-bs-dismiss="alert"
          aria-label="Close"
        ></button>
      </div>
      </div>
    </div>
  `;
};
