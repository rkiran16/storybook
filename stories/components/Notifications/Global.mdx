import { Canvas, Meta, Story } from "@storybook/addon-docs";

# GLOBAL NOTIFICATIONS

Banners take over the top of an interface to show general notifications for the product or system, not a specific task.  

Message
- Messages should be short and succinct.
- Example messages: 
   1) Inform users of site-wide issues (ex: site down for maintenance)
   2) Display warnings or errors that will directly impact users’ ability to
complete certain tasks (ex: license expired)
- Only show one banner at a time.

Sizing
- Full width of viewport

Placement
- Do not cover other content with a banner notification.
- Place banner directly below the main navigation task bar. 16px between banner and the breadcrumbs.
- Banners are not sticky and should scroll with the other content on the page.

Dismissal
- Include an ‘X’ in the upper right corner to allow users to close the banner.
- They persist until they are dismissed by the user or the user takes action that resolves the notification.
- Banners may persist across multiple sessions.

### Default

<Canvas>
    <Story id="components-notifications-global--default"></Story>
</Canvas>

### With Action

<Canvas>
      <Story id="components-notifications-global--default-with-action"></Story>
</Canvas>

### Error

<Canvas>
      <Story id="components-notifications-global--error"></Story>
</Canvas>

### Warning 

<Canvas>
      <Story id="components-notifications-global--warning"></Story>
</Canvas>

### Success

<Canvas>
      <Story id="components-notifications-global--success"></Story>
</Canvas>

### Info 

<Canvas>
      <Story id="components-notifications-global--information"></Story>
</Canvas>