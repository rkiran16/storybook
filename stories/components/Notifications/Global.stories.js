import { GlobalNotification } from "./Global";
import docs from "./Global.mdx";
import { html } from "lit-html";

export default {
  title: "Components/Notifications/Global",
  parameters: {
    zeplinLink:
      "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6303d1e19f494b0fe0a52218",
      docs: {
        page: docs,
      },
  },
  argTypes: {
    color: {
      name: "Color",
      control: { type: "select" },
      defaultValue: "black",
      options: ["black", "warning", "success", "info", "error"],
    },
    icon: {
      name: "Icon",
      control: { type: "select" },
      defaultValue: "default",
      options: ["none", "default", "warning", "success"],
      mapping: {
        none: false,
        default: "same",
        warning: "warning",
        success: "success",
      },
    },
  },
};

const Template = (args) => html`${GlobalNotification(args)}`;

export const Default = Template.bind({});
Default.args = {
  title: "Error!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  color: "black",
};
export const DefaultWithAction = Template.bind({});
DefaultWithAction.args = {
  title: "Lorem Ipsum!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  color: "black",
  href: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6303d1e19f494b0fe0a52218",
  linkTitle: "Link Action",
};
export const Error = Template.bind({});
Error.args = {
  title: "Error!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  color: "danger",
};
export const Warning = Template.bind({});
Warning.args = {
  title: "Warning!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  color: "warning",
  icon: "warning",
};
export const Success = Template.bind({});
Success.args = {
  title: "Success!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  color: "success",
  icon: "success",
};
export const Information = Template.bind({});
Information.args = {
  title: "Info!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  color: "info",
};

export const WithAction = Template.bind({});
WithAction.args = {
  title: "Error!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  href: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6303d1e19f494b0fe0a52218",
  linkTitle: "Link Action",
  color: "danger",
};
