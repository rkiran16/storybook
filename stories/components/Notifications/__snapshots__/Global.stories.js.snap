// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Notifications/Global Default 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <div
        class=\\"alert alert-custom bg-",
        " alert-dismissible d-flex align-items-center text-white rounded-0 global-notification\\"
        role=\\"alert\\"
      >
        <div class=\\"container d-flex position-relative\\">
        ",
        "
        <div>
          <strong class=\\"p-2\\">",
        "</strong>
          <span> ",
        "</span>
          <span class=\\"ms-2 text-white\\">
            ",
        "
          </span>
        </div>
        <button
          type=\\"button\\"
          class=\\"btn-close btn-close-white p-1\\" 
          data-bs-dismiss=\\"alert\\"
          aria-label=\\"Close\\"
        ></button>
      </div>
      </div>
    </div>
  ",
      ],
      "values": Array [
        "black",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons\\"> error_outline </span> ",
          ],
          "values": Array [],
        },
        "Error!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        null,
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Notifications/Global Default With Action 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <div
        class=\\"alert alert-custom bg-",
        " alert-dismissible d-flex align-items-center text-white rounded-0 global-notification\\"
        role=\\"alert\\"
      >
        <div class=\\"container d-flex position-relative\\">
        ",
        "
        <div>
          <strong class=\\"p-2\\">",
        "</strong>
          <span> ",
        "</span>
          <span class=\\"ms-2 text-white\\">
            ",
        "
          </span>
        </div>
        <button
          type=\\"button\\"
          class=\\"btn-close btn-close-white p-1\\" 
          data-bs-dismiss=\\"alert\\"
          aria-label=\\"Close\\"
        ></button>
      </div>
      </div>
    </div>
  ",
      ],
      "values": Array [
        "black",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons\\"> error_outline </span> ",
          ],
          "values": Array [],
        },
        "Lorem Ipsum!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        Object {
          "_$litType$": 1,
          "strings": Array [
            "<a class=\\"inline text-white\\" href=",
            ">",
            "</a>",
          ],
          "values": Array [
            "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6303d1e19f494b0fe0a52218",
            "Link Action",
          ],
        },
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Notifications/Global Error 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <div
        class=\\"alert alert-custom bg-",
        " alert-dismissible d-flex align-items-center text-white rounded-0 global-notification\\"
        role=\\"alert\\"
      >
        <div class=\\"container d-flex position-relative\\">
        ",
        "
        <div>
          <strong class=\\"p-2\\">",
        "</strong>
          <span> ",
        "</span>
          <span class=\\"ms-2 text-white\\">
            ",
        "
          </span>
        </div>
        <button
          type=\\"button\\"
          class=\\"btn-close btn-close-white p-1\\" 
          data-bs-dismiss=\\"alert\\"
          aria-label=\\"Close\\"
        ></button>
      </div>
      </div>
    </div>
  ",
      ],
      "values": Array [
        "danger",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons\\"> error_outline </span> ",
          ],
          "values": Array [],
        },
        "Error!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        null,
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Notifications/Global Information 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <div
        class=\\"alert alert-custom bg-",
        " alert-dismissible d-flex align-items-center text-white rounded-0 global-notification\\"
        role=\\"alert\\"
      >
        <div class=\\"container d-flex position-relative\\">
        ",
        "
        <div>
          <strong class=\\"p-2\\">",
        "</strong>
          <span> ",
        "</span>
          <span class=\\"ms-2 text-white\\">
            ",
        "
          </span>
        </div>
        <button
          type=\\"button\\"
          class=\\"btn-close btn-close-white p-1\\" 
          data-bs-dismiss=\\"alert\\"
          aria-label=\\"Close\\"
        ></button>
      </div>
      </div>
    </div>
  ",
      ],
      "values": Array [
        "info",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons\\"> error_outline </span> ",
          ],
          "values": Array [],
        },
        "Info!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        null,
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Notifications/Global Success 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <div
        class=\\"alert alert-custom bg-",
        " alert-dismissible d-flex align-items-center text-white rounded-0 global-notification\\"
        role=\\"alert\\"
      >
        <div class=\\"container d-flex position-relative\\">
        ",
        "
        <div>
          <strong class=\\"p-2\\">",
        "</strong>
          <span> ",
        "</span>
          <span class=\\"ms-2 text-white\\">
            ",
        "
          </span>
        </div>
        <button
          type=\\"button\\"
          class=\\"btn-close btn-close-white p-1\\" 
          data-bs-dismiss=\\"alert\\"
          aria-label=\\"Close\\"
        ></button>
      </div>
      </div>
    </div>
  ",
      ],
      "values": Array [
        "success",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons\\"> check_circle_outline </span> ",
          ],
          "values": Array [],
        },
        "Success!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        null,
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Notifications/Global Warning 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <div
        class=\\"alert alert-custom bg-",
        " alert-dismissible d-flex align-items-center text-white rounded-0 global-notification\\"
        role=\\"alert\\"
      >
        <div class=\\"container d-flex position-relative\\">
        ",
        "
        <div>
          <strong class=\\"p-2\\">",
        "</strong>
          <span> ",
        "</span>
          <span class=\\"ms-2 text-white\\">
            ",
        "
          </span>
        </div>
        <button
          type=\\"button\\"
          class=\\"btn-close btn-close-white p-1\\" 
          data-bs-dismiss=\\"alert\\"
          aria-label=\\"Close\\"
        ></button>
      </div>
      </div>
    </div>
  ",
      ],
      "values": Array [
        "warning",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons\\"> warning_amber </span> ",
          ],
          "values": Array [],
        },
        "Warning!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        null,
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Notifications/Global With Action 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <div
        class=\\"alert alert-custom bg-",
        " alert-dismissible d-flex align-items-center text-white rounded-0 global-notification\\"
        role=\\"alert\\"
      >
        <div class=\\"container d-flex position-relative\\">
        ",
        "
        <div>
          <strong class=\\"p-2\\">",
        "</strong>
          <span> ",
        "</span>
          <span class=\\"ms-2 text-white\\">
            ",
        "
          </span>
        </div>
        <button
          type=\\"button\\"
          class=\\"btn-close btn-close-white p-1\\" 
          data-bs-dismiss=\\"alert\\"
          aria-label=\\"Close\\"
        ></button>
      </div>
      </div>
    </div>
  ",
      ],
      "values": Array [
        "danger",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons\\"> error_outline </span> ",
          ],
          "values": Array [],
        },
        "Error!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        Object {
          "_$litType$": 1,
          "strings": Array [
            "<a class=\\"inline text-white\\" href=",
            ">",
            "</a>",
          ],
          "values": Array [
            "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6303d1e19f494b0fe0a52218",
            "Link Action",
          ],
        },
      ],
    },
  ],
}
`;
